Traceback (most recent call last):
  File "/home/runner/.cache/pypoetry/virtualenvs/quantflow-lUXkzsy2-py3.12/lib/python3.12/site-packages/jupyter_core/utils/__init__.py", line 154, in wrapped
    asyncio.get_running_loop()
RuntimeError: no running event loop

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/runner/.cache/pypoetry/virtualenvs/quantflow-lUXkzsy2-py3.12/lib/python3.12/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/home/runner/.cache/pypoetry/virtualenvs/quantflow-lUXkzsy2-py3.12/lib/python3.12/site-packages/nbclient/client.py", line 1319, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/.cache/pypoetry/virtualenvs/quantflow-lUXkzsy2-py3.12/lib/python3.12/site-packages/jupyter_core/utils/__init__.py", line 158, in wrapped
    return loop.run_until_complete(inner)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.12.11/x64/lib/python3.12/asyncio/base_events.py", line 691, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/home/runner/.cache/pypoetry/virtualenvs/quantflow-lUXkzsy2-py3.12/lib/python3.12/site-packages/nbclient/client.py", line 709, in async_execute
    await self.async_execute_cell(
  File "/home/runner/.cache/pypoetry/virtualenvs/quantflow-lUXkzsy2-py3.12/lib/python3.12/site-packages/nbclient/client.py", line 1062, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/home/runner/.cache/pypoetry/virtualenvs/quantflow-lUXkzsy2-py3.12/lib/python3.12/site-packages/nbclient/client.py", line 918, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
d = await cli.profile(stock)
d
------------------


[31m---------------------------------------------------------------------------[39m
[31mHttpResponseError[39m                         Traceback (most recent call last)
[36mCell[39m[36m [39m[32mIn[5][39m[32m, line 1[39m
[32m----> [39m[32m1[39m d = [38;5;28;01mawait[39;00m cli.profile(stock)
[32m      2[39m d

[36mFile [39m[32m~/work/quantflow/quantflow/quantflow/data/fmp.py:57[39m, in [36mFMP.profile[39m[34m(self, *tickers, **kw)[39m
[32m     55[39m [38;5;28;01masync[39;00m [38;5;28;01mdef[39;00m[38;5;250m [39m[34mprofile[39m([38;5;28mself[39m, *tickers: [38;5;28mstr[39m, **kw: Any) -> [38;5;28mlist[39m[[38;5;28mdict[39m]:
[32m     56[39m [38;5;250m    [39m[33;03m"""Company profile - minute"""[39;00m
[32m---> [39m[32m57[39m     [38;5;28;01mreturn[39;00m [38;5;28;01mawait[39;00m [38;5;28mself[39m.get_path([33mf[39m[33m"[39m[33mprofile/[39m[38;5;132;01m{[39;00m[38;5;28mself[39m.join(*tickers)[38;5;132;01m}[39;00m[33m"[39m, **kw)

[36mFile [39m[32m~/work/quantflow/quantflow/quantflow/data/fmp.py:217[39m, in [36mFMP.get_path[39m[34m(self, path, **kw)[39m
[32m    216[39m [38;5;28;01masync[39;00m [38;5;28;01mdef[39;00m[38;5;250m [39m[34mget_path[39m([38;5;28mself[39m, path: [38;5;28mstr[39m, **kw: Any) -> [38;5;28mlist[39m[[38;5;28mdict[39m]:
[32m--> [39m[32m217[39m     result = [38;5;28;01mawait[39;00m [38;5;28mself[39m.get([33mf[39m[33m"[39m[38;5;132;01m{[39;00m[38;5;28mself[39m.url[38;5;132;01m}[39;00m[33m/[39m[38;5;132;01m{[39;00mpath[38;5;132;01m}[39;00m[33m"[39m, **[38;5;28mself[39m.params(**kw))
[32m    218[39m     [38;5;28;01mreturn[39;00m cast([38;5;28mlist[39m[[38;5;28mdict[39m], result)

[36mFile [39m[32m~/.cache/pypoetry/virtualenvs/quantflow-lUXkzsy2-py3.12/lib/python3.12/site-packages/fluid/utils/http_client.py:164[39m, in [36mHttpClient.get[39m[34m(self, url, **kwargs)[39m
[32m    163[39m [38;5;28;01masync[39;00m [38;5;28;01mdef[39;00m[38;5;250m [39m[34mget[39m([38;5;28mself[39m, url: [38;5;28mstr[39m, **kwargs: Any) -> ResponseType:
[32m--> [39m[32m164[39m     [38;5;28;01mreturn[39;00m [38;5;28;01mawait[39;00m [38;5;28mself[39m.request([33m"[39m[33mGET[39m[33m"[39m, url, **kwargs)

[36mFile [39m[32m~/.cache/pypoetry/virtualenvs/quantflow-lUXkzsy2-py3.12/lib/python3.12/site-packages/fluid/utils/http_client.py:216[39m, in [36mHttpClient.request[39m[34m(self, method, url, headers, callback, monitor_http, **kw)[39m
[32m    214[39m     data = {}
[32m    215[39m [38;5;28;01melse[39;00m:
[32m--> [39m[32m216[39m     [38;5;28;01mawait[39;00m [38;5;28mself[39m.response_error(response)
[32m    217[39m [38;5;28;01mreturn[39;00m data

[36mFile [39m[32m~/.cache/pypoetry/virtualenvs/quantflow-lUXkzsy2-py3.12/lib/python3.12/site-packages/fluid/utils/http_client.py:234[39m, in [36mHttpClient.response_error[39m[34m(cls, response)[39m
[32m    232[39m [38;5;28;01mexcept[39;00m [38;5;167;01mException[39;00m:
[32m    233[39m     data = {[33m"[39m[33mmessage[39m[33m"[39m: [38;5;28;01mawait[39;00m response.text()}
[32m--> [39m[32m234[39m [38;5;28;01mraise[39;00m [38;5;28mcls[39m.ResponseError(response, data)

[31mHttpResponseError[39m: {
  "response": [],
  "request_url": "https://financialmodelingprep.com/stable/profile/KNOS.L?apikey=deb494ca540f0cbda22b0c38682115a3",
  "request_method": "GET",
  "response_status": 404
}

